src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs

```diff
@@ -309,6 +315,54 @@ private void ApplyDefaultWebViewSettings(BlazorWebViewDeveloperTools devTools)
 			// Desktop applications almost never want to show a URL preview when hovering over a link
 			_webview.CoreWebView2.Settings.IsStatusBarEnabled = false;
 		}
+
+#if WEBVIEW2_WINFORMS || WEBVIEW2_WPF
+		private static string GetUserDataFolderOverride()
+		{
+			if (Assembly.GetEntryAssembly() is { } mainAssembly)
+			{
+				// Where possible, and especially in development, we prefer not to override the WebView2 default behavior of creating
+				// its user data directory in the application folder. This helps to avoid littering the developer's user profile directory
+				// if they are repeatedly creating test applications.
+				if (!string.IsNullOrEmpty(mainAssembly.Location)
+					&& CanCreateSubdirectory(Path.GetDirectoryName(mainAssembly.Location)))
+				{
+					return null;
+				}
+
+				// However, if the application is running from a location where we can't write files (e.g., program files),
+				// use our own convention of %LocalAppData%\YourApplicationName.WebView2
+				var applicationName = mainAssembly.GetName().Name;
+				var result = Path.Combine(
+					Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+					$"{applicationName}.WebView2");
+
+				return result;
+			}
+
+			return null;
+		}
+
+		private static bool CanCreateSubdirectory(string withinDirectory)
+		{
+			if (!Directory.Exists(withinDirectory))
+			{
+				return false;
+			}
+
+			try
+			{
+				var path = Path.Combine(withinDirectory, Guid.NewGuid().ToString());
+				var dirInfo = Directory.CreateDirectory(path);
+				Directory.Delete(path, false);
+				return true;
+			}
+			catch (Exception)
```

Comment:
I don't like shipping code that expects exceptions. I'd rather pick a better default location and avoid this.